src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,27 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object r) {
+        if (this == r) {
+            return true;
+        }
+        if (r == null || getClass() != r.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) r;
+        return width.equals(rectangle.width)
+                && length.equals(rectangle.length)
+                && color.equals(rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 13;
+        result *= 31 + width.hashCode();
```

Comment:
what happens, if one of the fields == null ?

Reply:
Checking for null was added